### PR TITLE
Add security consideration for DoS vector

### PR DESCRIPTION
**Pull request by SulemanAhmadd**
Addresses issue #413

### CODE REVIEW

**Comment by simon-friedberger** [comment]
I think this is too generic to be very useful. It would be better to include this in the VDAF spec next to Poplar1 and mention the specific optimization of early-abort for short URLs.

**Comment by cjpatton** [comment]
I agree with @simon-friedberger: A more concrete example would help clarify.

**Comment by SulemanAhmadd** [comment]
Thank you folks for the suggestions. I have expanded the scope of the text by adding a representable example but keeping it generic enough to be broadly applicable. I believe, since DAP is the driver for the execution of the VDAF, this consideration seems more relevant to the DAP draft but I am open to redirections.

File: draft-ietf-ppm-dap.md [hunk 1]
```diff
@@ -2235,6 +2235,25 @@ Details for each issue are below.
...
+   configured to compute a set of heavy hitters requires each measurement to
+   be of the same bit-length which all parties need to agree on prior to VDAF
+   execution. The computation required for such tasks can increase
+   superlinearly as multiple rounds of evaluation are needed for each bit of
+   the measurement value.
+   When dealing with variable length inputs (e.g domain names), it might be
+   necessary to pad them to convert into fixed-size measurements. When
+   computing the heavy hitters from a batch of such measurements, we can
+   early-abort the Poplar1 execution once we have reached the padding region
+   for a candidate measurement. For smaller length inputs, it significantly
```
Comment (cjpatton) [inline draft-ietf-ppm-dap.md:2249]:
```suggestion
   for a candidate measurement. For smaller length inputs, this significantly
```

File: draft-ietf-ppm-dap.md [hunk 2]
```diff
@@ -2235,6 +2235,25 @@ Details for each issue are below.
...
+   reduces the cost of communication between aggregators and the steps
```
Comment (cjpatton) [inline draft-ietf-ppm-dap.md:2250]:
```suggestion
   reduces the cost of communication between Aggregators and the steps
```

File: draft-ietf-ppm-dap.md [hunk 3]
```diff
@@ -2235,6 +2235,25 @@ Details for each issue are below.
...
+   required for the computation. However, malicious clients can still generate
```
Comment (cjpatton) [inline draft-ietf-ppm-dap.md:2251]:
```suggestion
   required for the computation. However, malicious Clients can still generate
```

**Review by cjpatton — APPROVED** [review]
Looks good, just nits!

**Comment by SulemanAhmadd** [comment]
Thanks, comments have been addressed.

**Review by tgeoghegan — APPROVED** [review]

**Pull request merged**